File: 209. Minimum Size Subarray Sum/solution.go
package main

import "fmt"

func twoPointers(target int, nums []int) int {
	i, res := 0, len(nums)+1
	for j := 0; j < len(nums); j++ {
		target -= nums[j]
		for target <= 0 {
			if res > j-i+1 {
				res = j - i + 1
			}
			target += nums[i]
			i++
		}
	}
	return res % (len(nums) + 1)
}

func main() {
	target := 7
	nums := []int{2, 3, 1, 2, 4, 3}
	fmt.Println(twoPointers(target, nums))
}

File: 209. Minimum Size Subarray Sum/solution.py
class Solution:
    def minSubArrayLen(self, target: int, nums: list[int]) -> int:
        l = 0
        r = 0
        now = 0
        ans = float("INF")
        while r < len(nums):
            now += nums[r]
            r += 1
            while now >= target:
                ans = min(ans, r - l)
                now -= nums[l]
                l += 1
        if ans == float("INF"):
            return 0
        return ans
